ui/src/kapacitor/components/RuleGraph.tsx

```diff
@@ -19,13 +20,15 @@ import {LINE_COLORS_RULE_GRAPH} from 'src/shared/constants/graphColorPalettes'
 import {Source, AlertRule, QueryConfig, Query, TimeRange} from 'src/types'
 
 import {ErrorHandling} from 'src/shared/decorators/errors'
+import {setHoverTime as setHoverTimeAction} from 'src/dashboards/actions'
 
 interface Props {
   source: Source
   query: QueryConfig
   rule: AlertRule
   timeRange: TimeRange
   onChooseTimeRange: (tR: TimeRange) => void
+  setHoverTime: (time: number) => void
```

Comment:
what do you think about type-annotating this via `typeof <action>` (as seen in `DashboardPage.tsx`) so that the interface expectation remains in sync with its target action?

Reply:
love it. doing it.